### PR TITLE
Connecting Announcement System with firebase storage

### PR DESCRIPTION
**Pull request by reyimanuel**
# **Connecting Announcement System with firebase storage**

Updating the functionality on crud system on the repositories file, user with role 'Pengurus' had the privilege to use post, put, delete method while the user with role 'Anggota' can only use the get method

## **Deskripsi**

**Endpoint Pengurus**
1. Show:

image

2. Create:

image

3. Update:
3.1 update user_id = creator_id

image

3.2 update user_id != creator_id

image

note: As for the method put isn't usable with form data, and somehow using post with put parameter works so that does it ig
logic: If user trying change an image from a data, it will delete the old image file from firebase and changing it to a new one
4. Delete
4.1 delete user_id = creator_id

image

4.2 delete user_id != creator_id

image

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -45,8 +45,12 @@ function () {
         Route::get('/pengurus', function () {
             return response()->json(['message' => 'Selamat Datang Pengurus', 'user' => $GLOBALS['USER_DATA']->name]);
         });
+        Route::post('/announcements', [AnnouncementController::class, 'store']);
+        Route::put('/announcements/{announcement}', [AnnouncementController::class, 'update']);
+        Route::delete('/announcements/{announcement}', [AnnouncementController::class, 'destroy']);
     });
-
+    Route::get('/announcements', [AnnouncementController::class, 'index']);
+    Route::get('/announcements/{announcement}', [AnnouncementController::class, 'show']);
```
Comment (zharkaa) [inline routes/api.php:53]: untuk endpoint `GET` ditempatkan di user dengan role `member/anggota`.

untuk skarang kse pindah jo ini endpoint.

**Review by zharkaa — CHANGES_REQUESTED** [review]
silahkan diperbaiki

note: dpe nama PR jgn bgini neh wkwkwk sesuaikan dpe nama dengan apa ini PR ada bking

cc @reyimanuel

File: routes/api.php (hunk 1 above)
Comment (reyimanuel) [reply routes/api.php:53]: woke

Comment (reyimanuel) [reply routes/api.php]: don ya kak 🙏

**Review by zharkaa — APPROVED** [review]
LGTM!

**Pull request merged**